src/core/createConfig.js

```diff
@@ -76,7 +76,11 @@ const createConfig = config => {
             validator.defaultValue
           );
           if (errorMessage) {
-            ac.push(errorMessage);
+            if (isString(errorMessage)) {
+              ac.push(errorMessage);
+            } else {
+              ac.push(`An internal error occurred while validating '${key}'`);
```

Comment:
Should you log the faulty errorMessage here just so we could debug the issue if needed?

Reply:
@jonsnyder thinking through this again, would be better to just add tests for our config validators (e.g, validate that `optInEnabled` accepts a boolean and only a boolean)?